languagetool-language-modules/en/src/main/resources/org/languagetool/rules/en/grammar.xml

```diff
@@ -83988,6 +83988,29 @@ USA
             <example correction="must|have to">I <marker>must to</marker> go shopping.</example>
             <example>It is a must to be aware of that.</example>
         </rule>
+        <rule id="TO_AFTER_MODAL_VERBS" name="to after modal verbs" default="temp_off">
+         <antipattern>
+            <token regexp="yes">an?</token>
+            <token regexp="yes" min="0">absolute|total</token>
+            <token regexp="yes">can|could|may|might|should|shall</token>
+         </antipattern>
+         <antipattern>
+            <token regexp="yes">that|(what|which|who)(ever)?</token>
+            <token chunk_re="B-NP.*" min="0" />
+            <token chunk_re="E-NP.*" />
+            <token postag="RB" min="0" />
+            <token regexp="yes">can|could|may|might|should|shall</token>
+            <example>I will do whatever I can to help.</example>
+         </antipattern>
+         <pattern>
+             <token regexp='yes' postag='MD'>can|could|may|might|should|shall</token>
+             <token>to</token>
+         </pattern>
+         <message>Modal verbs are followed directly by the infinitive of another verb, without 'to'. Probably you should use <suggestion>base form</suggestion></message>
```

Comment:
This suggestion will suggest the literal phrase _base form_. 
There are a couple of ways to suggest what you want. What I would probably do: 
Extend the pattern:
```
    <pattern>
        <token regexp='yes' postag='MD'>can|could|may|might|should|shall</token>
        <token>to</token>
        <token postag="VB"/>
    </pattern>
```
And for the suggestion: `<suggestion>\1 \3</suggestion>`

Don't forget, then to also modify the `example correction` and the `marker` in it. 

But there are various approaches here.